src/plugin/Plugin.h

```diff
@@ -113,13 +116,82 @@ class Configuration
 	// We force this to inline so that the API version gets hardcoded
 	// into the external plugin. (Technically, it's not a "force", just a
 	// strong hint.). The attribute seems generally available.
-	inline Configuration() __attribute__((always_inline)) { bro_version = BRO_PLUGIN_BRO_VERSION; }
+	inline Configuration() __attribute__((always_inline))
+		{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+		bro_version = ZEEK_PLUGIN_ZEEK_VERSION;
+#pragma GCC diagnostic pop
+		zeek_version = ZEEK_PLUGIN_ZEEK_VERSION;
+		}
+
+	Configuration(Configuration&& c)
+		{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+		bro_version = std::move(c.bro_version);
+#pragma GCC diagnostic pop
+		zeek_version = std::move(c.zeek_version);
+
+		name = std::move(c.name);
+		description = std::move(c.description);
+		version = std::move(c.version);
+		}
+
+	Configuration(const Configuration& c)
+		{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+		bro_version = c.bro_version;
+#pragma GCC diagnostic pop
+		zeek_version = c.zeek_version;
+
+		name = c.name;
+		description = c.description;
+		version = c.version;
+		}
+
+	Configuration& operator=(Configuration&& c)
+		{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+		bro_version = std::move(c.bro_version);
+#pragma GCC diagnostic pop
+		zeek_version = std::move(c.zeek_version);
+
+		name = std::move(c.name);
+		description = std::move(c.description);
+		version = std::move(c.version);
+
+		return *this;
+		}
+
+	Configuration& operator=(const Configuration& c)
+		{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+		bro_version = c.bro_version;
+#pragma GCC diagnostic pop
+		zeek_version = c.zeek_version;
+
+		name = c.name;
+		description = c.description;
+		version = c.version;
+
+		return *this;
+		}
+
+	/**
+	 * One can assign ZEEK_PLUGIN_ZEEK_VERSION to this to catch
+	 * version mismatches at link(!) time.
+	 */
+	[[deprecated("Remove in v6.1. Use zeek_version.")]] std::function<const char*()> bro_version;
 
 	/**
-	 * One can assign BRO_PLUGIN_BRO_VERSION to this to catch
+	 * One can assign ZEEK_PLUGIN_ZEEK_VERSION to this to catch
 	 * version mismatches at link(!) time.
 	 */
-	const char* (*bro_version)();
+	std::function<const char*()> zeek_version;
```

Comment:
Do we need the new methods above because of the change to `std::function`? If so, I'm wondering if we should just stay with the raw function pointer just for brevity.

Reply:
No, the same thing happens even without `std::function`. You get an error from the compiler about ambiguity with the `move` constructor specifically, but adding just that causes other errors to pop up.